src/hooks/use-payment-status.ts

```
@@ -0,0 +1,46 @@
+  const maxRetries = 3
+
+  useEffect(() => {
+    let retryTimeout: ReturnType<typeof setTimeout>
+    const eventSource = new EventSource(`/api/payments/sse?payment_id=${paymentId}`)
+
+    eventSource.onmessage = (event) => {
+      try {
+        const data = JSON.parse(event.data)
+        setPayment(data)
+        setRetryCount(0) // Reset retry count on successful message
+      } catch (error) {
```

ESLint

[error] Closing curly brace appears on the same line as the subsequent block.

(style/brace-style)